### PR TITLE
[ML] Make coefficient of variation formula more robust

### PR DESCRIPTION
**Pull request by droberts195**
The previous formula was vulnerable to dividing by zero
in unlikely edge cases.  This change rearranges the formula
to avoid that, and also adds a test to prove no crashes
occur in edge cases.

Relates #1623

### CODE REVIEW

**Comment by droberts195** [comment]
`>non-issue` because logically this is part of #1623

File: lib/model/CResourceMonitor.cc [hunk 1]
```diff
@@ -330,12 +338,14 @@ bool CResourceMonitor::isMemoryStable(core_t::TTime bucketLength) const {
...
-              << maths::CBasicStatistics::count(m_ModelBytesMoments) << ", mean = " << mean
-              << ", variance = " << variance << ", coefficient of variation = " << cv);
-    if (cv > ESTABLISHED_MEMORY_CV_THRESHOLD) {
+              << ": bucket count = " << bucketCount
+              << ", sample count = " << maths::CBasicStatistics::count(m_ModelBytesMoments)
+              << ", mean = " << mean << ", variance = " << variance
+              << ", coefficient of variation = " << (std::sqrt(variance) / mean));
+    // Instead of literally testing the coefficient of variation it's more
+    // robust against zeroes and NaNs to rearrange it as follows
+    if (variance > maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean)) {
```
Comment (tveasey) [inline lib/model/CResourceMonitor.cc:348]: Regarding your comment on NaNs: is it actually better to return `false` in the case of NaN? Perhaps it is. Unlike 754, C++ doesn't guaranty comparison operator results on NaNs as I recall, so should we replace
```
if (variance > maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean)) {
    return false;
}

return true;
```
with
```
return CMathsFuncs::isNan(variance) &&
       variance <= maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean);
```
Even if we don't explicitly check NaN, it feels slightly cleaner not to bother with the `if`.

**Review by tveasey — APPROVED** [review]
Just a minor suggestion/question regarding NaN. Otherwise, LGTM. Also, no need for another review.
